lib/amq/uri.rb

```diff
@@ -6,7 +6,10 @@
 module AMQ
   class URI
```

Comment:
What do you think if we try to conform `URI` Ruby standard library hierarchy and extract separate `AMP::URI::AMQP` and `AMP::URI::AMQPS` classes?

![amq uri](https://user-images.githubusercontent.com/892721/33828560-4d9d97f4-de75-11e7-94e2-dc6a27be11d7.png)

Reply:
I don't see any real benefit.

Reply:
It could simplify the code and make it more object-oriented instead of procedural in my opinion. Because AMQP URI is just another kind of generic URI, contributors, that are familiar already with `URI` (especially as it's a standard library), intuitively could rely on known public API.

Reply:
I don't see how having 4 new classes would simplify things. We are not preparing this library for inclusion into the standard library.

Moving classes and constants around means Bunny and amqp gem will have to be updated, and their older versions won't be compatible with newer version of amq-protocol, which has been the case for years.

Reply:
Sorry but "more object-oriented code" for a couple of constants is a solution in search of a problem.

Reply:
I see, appreciate your open point of view, thank you 🙇